### PR TITLE
Use a region specific federated url per the AWS docs.

### PR DESCRIPTION
**Pull request by synfinatic**
Need user testing to confirm it works in china, but it seems to still work in us-east-2 :)

Fixes: #634

### CODE REVIEW

**Comment by deanmax** [comment]
did a local build, still seeing the error thou
```
❯ dist/aws-sso-1.14.2-darwin-amd64 --sso=china console --account=112233445566 --role=CHINA-DEVOPS --duration=720
FATAL   Error running command: Error parsing Login response: invalid character '<' looking for beginning of value
```

Also verified us-west-2, which works fine
```
❯ dist/aws-sso-1.14.2-darwin-amd64 console --sso=default --account=1122334455 --role=sso-devops --duration=720
```

File: internal/url/url.go [hunk 1]
```diff
@@ -327,3 +327,20 @@ func SSOAuthAction(action Action) Action {
...
+
+const AWS_FEDERATED_URL_FORMAT = "https://%s.signin.%s/federation"
+
+// AWSFederatedUrl generates the region/partition specific URL for the AWS
+// Federated endpoint for IAM Identity Center
+// https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_enable-console-custom-url.html
+func AWSFederatedUrl(region string) string {
+	if strings.HasPrefix(region, "cn-") {
+		// china
+		return fmt.Sprintf(AWS_FEDERATED_URL_FORMAT, region, "amazonaws.com.cn")
```
Comment (deanmax) [inline internal/url/url.go:339]:
```suggestion
		return fmt.Sprintf(AWS_FEDERATED_URL_FORMAT, region, "amazonaws.cn")
```
Command works after making this change. However, in the browser window i'm still seeing this
Screenshot 2023-10-19 at 16 11 14

**Comment by synfinatic** [comment]
@deanmax This last change do it then I think.  Mind giving it a try?

**Comment by synfinatic** [comment]
@deanmax Doh.  Last change was bad.  This one is better and now it actually does DNS lookups in my tests to make sure I don't generate invalid FQDN's.

**Comment by deanmax** [comment]
Works great now! Thank you!

**Comment by synfinatic** [comment]
@deanmax gonna get this out now.

**Comment by synfinatic** [comment]
@deanmax release out.  If you're using brew:  https://github.com/Homebrew/homebrew-core/pull/151890

**Pull request merged**